src/docfx/restore/RestoreUrl.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Microsoft.Docs.Build
+{
+    public static class RestoreUrl
+    {
+        private const int MaxVersionCount = 5;
+
+        public static string GetRestoreVersionPath(string restoreDir, string version)
+            => PathUtility.NormalizeFile(Path.Combine(restoreDir, version));
+
+        public static string GetRestoreRootDir(string address)
+            => Docs.Build.Restore.GetRestoreRootDir(address, AppData.UrlRestoreDir);
+
+        public static async Task<string> Restore(string docset, string address)
+        {
+            var tempFile = await DownloadToTempFile(address);
+
+            var fileVersion = "";
+            using (var fileStream = File.Open(tempFile, FileMode.Open, FileAccess.Read))
+            {
+                fileVersion = PathUtility.Encode(HashUtility.GetMd5String(fileStream));
+            }
+
+            Debug.Assert(!string.IsNullOrEmpty(fileVersion));
+
+            var restoreDir = GetRestoreRootDir(address);
+            var restorePath = GetRestoreVersionPath(restoreDir, fileVersion);
```

Comment:
>restorePath [](start = 16, length = 11)

we don't need to prefix the download location with urls, the hash code is enough to identify a file.

Reply:
i want the file to be easily found manually, hash code is really hard for us for debuging.

---
In reply to: [202255047](https://github.com/dotnet/docfx/pull/3065#discussion_r202255047) [](ancestors = 202255047)